builder/vmware/iso/step_create_vmx.go

```diff
@@ -263,6 +266,18 @@ func (s *stepCreateVMX) Run(ctx context.Context, state multistep.StateBag) multi
 		templateData.Serial_Host = ""
 		templateData.Serial_Auto = "FALSE"
 
+		switch config.HWConfig.Firmware {
+		case "bios":
```

Comment:
I was wondering if the values of allowedFirmwareTypes were reused later on. It looks like  they are used here but they are hardcoded values. 

What do you think about  replacing `var allowedFirmwareTypes` with three constants one per type that you can reuse here? 

```go
const (
 FirmwareTypeBios = "bios"
 FirmwareTypeUEFI= "uefi"
 FirmwareTypeUEFISecure = "refi-secure"
)
```

With this constant in place you can rewrite the switch statement here to use the constants and in hw_config.go you can create the allowedFirmwareTypes slice local to where it is used on line 120

```go
allowedFirmwareTypes := []string{FirmwareTypeBios, FirmwareTypeUEFI, FirmwareTypeUEFISecure}
if (c.Firmware != "") && (slices.Contains(allowedFirmwareTypes, c.Firmware) == false) {
//.... 
}
```

Reply:
This is a great idea! I'll make it so. 👍

Reply:
Resolved in https://github.com/hashicorp/packer-plugin-vmware/commit/45c037f1538cda663def623f50ee7862c778bd70. Much cleaner approach.